test/Elastic.Apm.Tests/HttpDiagnosticListenerTests.cs

```diff
@@ -84,7 +84,7 @@ public void OnErrorLog()
 		/// Builds an HttpRequestMessage and calls HttpDiagnosticListener.OnNext directly with it.
 		/// Makes sure that the processingRequests dictionary captures the ongoing transaction.
 		/// </summary>
-		[Fact]
+		[NetCoreFact] //see: https://github.com/elastic/apm-agent-dotnet/issues/516
```

Comment:
Could you please clarify what are the advantages of using runtime condition (via `[NetCoreFact]`) to skip tests vs. using `#if NETCOREAPP` (https://github.com/elastic/apm-agent-dotnet/pull/517/commits/d2813dac06bb808479aefe02572c02a0c020039d)? The advantage of using `#if NETCOREAPP` is that .NET Core only tests won't show as skipped for .NET Framework - the tests won't show at all.

It's worth discussing when tests should be marked as skipped. I think we should use skipped to mean that some preconditions **can be** but are not satisfied. If the preconditions cannot be satisfied then tests should not be included at all. For example running Full Framework tests on Linux should not include them at all while running Full Framework tests on Windows without environment variable `ELASTIC_APM_TESTS_FULL_FRAMEWORK_ENABLED` set should include the tests and mark them as skipped. This way a developer knows that if there are skipped tests that means they don't have some components in their environment to support all the available tests.

Reply:
I usually try to avoid `#if`s when possible. One benefit is having the same binary both for .NET Core and Full Framework - the more `#if`s we have the bigger the difference is between those two binaries. Also I think it’s harder to read that code, since you have to keep in mind on which .NET Flavor you currently are. 

I’d prefer having runtime checks - I think it’s better to have skipped tests with a proper explanation (in this case the `NetCoreFact` has it) than having tests that disappear. Also to me this seems to be a simpler option.

So I don’t see having skipped tests as a big problem - I’d say if from some reason a test cannot be executed it should be skipped but not hidden.